crates/bevy_ecs/src/component.rs

```diff
@@ -14,26 +14,103 @@ use std::{
     mem::needs_drop,
 };
 
-/// A component is data associated with an [`Entity`](crate::entity::Entity). Each entity can have
-/// multiple different types of components, but only one of them per type.
+/// A data type that can be used to store data for an [entity].
 ///
-/// Any type that is `Send + Sync + 'static` can implement `Component` using `#[derive(Component)]`.
 ///
-/// In order to use foreign types as components, wrap them using a newtype pattern.
+/// `Component` is a [derivable trait]: this means that a data type can implement it by simply prepending a `#[derive(Component)]` attribute to it.
+/// However, components must always satisfy the `Send + Sync + 'static` trait bounds.
+///
+/// [entity]: crate::entity
+/// [derivable trait]: https://doc.rust-lang.org/book/appendix-03-derivable-traits.html
+/// [`World`]: crate::world::World
+///
+/// # Examples
+///
+/// Components can take many forms: they are usually structs, but can also be of every other kind of data type, like enums or zero sized types.
+/// The following examples show how components are laid out in code.
+///
+/// ```
+/// # use bevy_ecs::component::Component;
+/// # struct Color;
+/// #
+/// // A component can contain data...
+/// #[derive(Component)]
+/// struct LicensePlate(String);
+///
+/// // ... but it can also be a zero-sized marker.
+/// #[derive(Component)]
+/// struct Car;
+///
+/// // Most commonly, components are structs with a small number of fields.
```

Comment:
I don't think specifying "Most commonly" or "less commonly" is really useful, apart from forcing a restraint on users

Reply:
Resolved by https://github.com/bevyengine/bevy/pull/4767/commits/d956aeadc37ef4b7398c5c64a53ca483d7352233